deepsensor/model/model.py

```diff
@@ -811,147 +835,265 @@ def get_index(*args, x1 = True) -> Union[int, Tuple[List[int], List[int]]]:
+                        patch_clip_x1_max = int(
+                            data_array.sizes[unnorm_coord_names["x1"]] - b_x1_max
+                        )
                         patch_clip_x2_min = int(b_x2_min)
-                        patch_clip_x2_max = int(data_array.sizes['x'] - b_x2_max)
+                        patch_clip_x2_max = int(
+                            data_array.sizes[unnorm_coord_names["x2"]] - b_x2_max
+                        )
 
-                        patch_clip = data_array.isel(y=slice(patch_clip_x1_min, patch_clip_x1_max),
-                                                    x=slice(patch_clip_x2_min, patch_clip_x2_max))
+                        patch_clip = data_array[
+                            {
+                                unnorm_coord_names["x1"]: slice(
+                                    patch_clip_x1_min, patch_clip_x1_max
+                                ),
+                                unnorm_coord_names["x2"]: slice(
+                                    patch_clip_x2_min, patch_clip_x2_max
+                                ),
+                            }
+                        ]
 
                         patches_clipped[var_name].append(patch_clip)
 
-            combined = {var_name: xr.combine_by_coords(patches, compat='no_conflicts') for var_name, patches in patches_clipped.items()}
+            combined = {
+                var_name: xr.combine_by_coords(patches, compat="no_conflicts")
+                for var_name, patches in patches_clipped.items()
+            }
 
             return combined
 
+        # sanitise patch_size and stride arguments
+
+        if isinstance(patch_size, float) and patch_size is not None:
+            patch_size = (patch_size, patch_size)
+
+        if isinstance(stride, float) and stride is not None:
+            stride = (stride, stride)
+
+        if stride[0] > patch_size[0] or stride[1] > patch_size[1]:
+            raise ValueError(
+                f"stride must be smaller than patch_size in the corresponding dimensions. Got: patch_size: {patch_size}, stride: {stride}"
+            )
+
+        for val in list(stride + patch_size):
+            if val > 1.0 or val < 0.0:
+                raise ValueError(
+                    f"Values of stride and patch_size must be between 0 & 1. Got: patch_size: {patch_size}, stride: {stride}"
+                )
+
+        # Get coordinate names of original unnormalised dataset.
+        unnorm_coord_names = {
+            "x1": self.data_processor.raw_spatial_coord_names[0],
+            "x2": self.data_processor.raw_spatial_coord_names[1],
+        }
+
+        # tasks should be iterable, if only one is provided, make it a list
+        if type(tasks) is Task:
+            tasks = [tasks]
+
         # Perform patchwise predictions
         preds = []
         for task in tasks:
-            bbox = task['bbox']
+            bbox = task["bbox"]
```

Comment:
Maybe add 'generated using sample_sliding_window(), as bbox is not produced for sample_random_window. We should also mention elsewhere that this code will not work if patching is done using random instead of sliding window.

Reply:
This only works for tasks generated using a patch_strategy of "sliding_window" and not "random".